src/inspector_agent.h

```diff
@@ -31,8 +46,27 @@ class InspectorSessionDelegate {
                                      = 0;
 };
 
-class InspectorIo;
-class NodeInspectorClient;
+class DispatcherSession {
+ public:
+  explicit DispatcherSession(ChannelImpl* channel) : channel_(channel) {}
+  ~DispatcherSession();
+  void SendMessageToFrontend(const v8_inspector::StringView& message);
+ private:
+  ChannelImpl* channel_;
+};
+
+class MessageDispatcher {
+ public:
+  virtual ~MessageDispatcher() = default;
+  virtual bool HandleMessage(const v8_inspector::StringView& message) = 0;
+};
+
+class MessageDispatcherFactory {
```

Comment:
Similarly to my recommendation below for `JSDispatcher`, this interface could be nested into the `MessageDispatcher` interface.